### PR TITLE
test: loosen metadata error check to reduce dependence on exact library errors

### PR DESCRIPTION
**Pull request by dfawley**
RELEASE NOTES: n/a

### CODE REVIEW

File: test/metadata_test.go [hunk 1]
```diff
@@ -113,7 +114,7 @@ func (s) TestInvalidMetadata(t *testing.T) {
 		if err := stream.Send(&testpb.StreamingOutputCallRequest{}); err != nil {
 			t.Errorf("call ss.Client stream Send(nil) will success but got err :%v", err)
 		}
-		if _, err := stream.Recv(); !reflect.DeepEqual(test.recv, err) {
+		if _, err := stream.Recv(); status.Code(err) != status.Code(test.recv) || !strings.Contains(err.Error(), test.recv.Error()) {
```
Comment (easwars) [inline test/metadata_test.go:117]: Should we switch the `reflect.DeepEqual()` on line 100 to something similar to here?

Comment (dfawley) [reply test/metadata_test.go:117]: Those errors are produced by our metadata library, so should be reliable.

Comment (easwars) [reply test/metadata_test.go:117]: What I meant was to get rid of the `DeepEqual()` and instead rely on direct comparison of the status code and the error string.

**Review by easwars — APPROVED** [review]

File: test/metadata_test.go (hunk 1 above)
Comment (dfawley) [reply test/metadata_test.go:117]: The problem is not the `DeepEqual` but the fact that we're testing for an error string that comes out of the standard library.  I think the `DeepEqual` is fine and simpler than checking the components separately.

**Pull request merged**